src/AudioAnalysisTools/Tracks/UpwardTrackAlgorithm.cs

```diff
@@ -155,7 +155,7 @@ public static Track GetUpwardTrack(double[,] peaks, int startRow, int startBin,
             for (int bin = startBin + 1; bin < maxBin - 1; bin++)
             {
                 // Avoid row edge effects.
-                if (row < 1 || row > peaks.GetLength(0) - 1)
+                if (row < 2 || row > peaks.GetLength(0) - 3)
```

Comment:
It'd be nice if these magic numbers were documented and explained

Reply:
They are due to edge effects of how the tracks are found and pushed upwards in this case.

Reply:
I have noted in some places that these adjustments are due to edge effects.